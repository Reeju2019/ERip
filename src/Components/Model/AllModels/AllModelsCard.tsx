import * as React from 'react'

interface IAllModelsCardProps {
  models: { model_name: string; image: string }
}

const AllModelsCard: React.FunctionComponent<IAllModelsCardProps> = (models) => {
  return (
    <>
      <div className='modelCard'>
        <div>
          <img src={models.models.image} className='modelImg' alt={models.models.model_name} />
        </div>
      </div>
      <p className='modelName'>{models.models.model_name}</p>
    </>
  )
}

export default AllModelsCard
