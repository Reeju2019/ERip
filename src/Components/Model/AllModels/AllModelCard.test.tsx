import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import AllModelsCard from './AllModelsCard'
import { MemoryRouter as Router } from 'react-router-dom'

interface Data {
    id: string
    modelName: string
    image: string
}
const data: Data = {
    id: "",
    modelName: "",
    image: ""
}
test('all model card', () => {
    render(<Router><AllModelsCard models={data} /></Router>)
})
describe('all model rendering specification', () => {
    it('SnapShot testing', () => {
        const component = renderer.create(<Router><AllModelsCard models={data} /></Router>)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('render without crashing all model', () => {
        const div = document.createElement('div')
        render(<Router><AllModelsCard models={data} /></Router>)
        ReactDOM.unmountComponentAtNode(div)
    })
})
