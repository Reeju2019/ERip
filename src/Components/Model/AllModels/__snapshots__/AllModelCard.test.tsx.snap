// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all model rendering specification SnapShot testing 1`] = `
Array [
  <div
    className="modelCard"
    onClick={[Function]}
  >
    <div>
      <img
        alt=""
        className="modelImg"
        src=""
      />
    </div>
  </div>,
  <p
    className="modelName"
  />,
]
`;
